pkg/cmd/gitprovider/delete.go

```diff
@@ -34,20 +36,25 @@ var gitProviderDeleteCmd = &cobra.Command{
 		var gitProviderData apiclient.SetGitProviderConfig
 		gitProviderData.BaseApiUrl = new(string)
 		gitProviderData.Username = new(string)
+		gitProviderData.Alias = new(string)
 
 		if len(gitProviders) == 0 {
 			views.RenderInfoMessage("No git providers registered")
 			return
 		}
 
-		gitprovider_view.GitProviderSelectionView(&gitProviderData, gitProviders, true)
+		gitprovider_view.GitProviderSelectionView(&gitProviderData, gitProviders, true, apiClient, ctx)
+		Id := &strings.Split(gitProviderData.ProviderId, "^")[0]
```

Comment:
Why is the ID fetched this way?

Reply:
But why don't we just use the `Id` in `gitProviderData`

Reply:
I get what you're trying to say, but you would still print out the ProviderId but the Id can come from `gitProviderData.Id` still because it's already there. No need to parse this.

Reply:
No, I'm saying that:
```golang
type GitProviderView struct {
	Id         string
	Name       string
	Username   string
	BaseApiUrl string
	Token      string
}
```

Should still have the ID there as well and just use that instead of `Id = ....`

Reply:
But the selection field must be the `Id` since multiple git providers can have the same `ProviderId`. That's what I'm saying. You need to change `Value(&gitProviderAddView.ProviderId))` to `Value(&gitProviderAddView.Id))`.

Reply:
Since you have the list of gitproviders from the API, you can find the right one and then display `Git Provider {gitProvider.ProviderId ({gitProvider.Alias}) has been removed`.

Reply:
No 😄 

Please test the code you're submitting before requesting a review. As I said before, since multiple git providers can have the same providerId, you can not use that to search through them. The selection view must set the appropriate Id (not ProviderId) on the `gitProviderData` object.

Reply:
No 😄 

As I said, `Id` needs to be set on `gitProviderData` **NOT** ProviderId. Please fix that in the selection view.

Reply:
Okay, this will make more sense. I'll resolve this once I test it more. Thanks

Reply:
> I'll resolve this once I test it more!

Please stop tagging us. I can't keep saying it.

Reply:
It probably will yes.